src/util/ActiveShapeUtils.tsx

```diff
@@ -63,3 +78,140 @@ export function Shape<OptionType, ExtraProps, ShapePropsType>({
 
   return shape;
 }
+
+/**
+ * This is an abstraction to handle identifying the active index from a tooltip mouse interaction
+ */
+type GraphicalItemShapeKey = 'trapezoids' | 'sectors' | 'points';
+
+type ItemDisplayName = 'Funnel' | 'Pie' | 'Scatter';
+
+type FunnelItem = {
+  x: number;
+  y: number;
+  labelViewBox: { x: number; y: number };
+  tooltipPayload: Array<{ payload: { payload: unknown } }>;
+};
+
+type PieItem = {
+  startAngle: number;
+  endAngle: number;
+  tooltipPayload: Array<{ payload: { payload: unknown } }>;
+};
+
+type ScatterItem = {
+  x: number;
+  y: number;
+  z: number;
+  payload?: unknown;
+};
+
+type ShapeData<DisplayName extends ItemDisplayName> = DisplayName extends 'Funnel'
+  ? FunnelItem
+  : DisplayName extends 'Pie'
+  ? PieItem
+  : DisplayName extends 'Scatter'
+  ? ScatterItem
+  : never;
+
+type GetActiveShapeIndexForTooltip = {
+  itemDisplayName: ItemDisplayName;
+  activeTooltipItem: ShapeData<ItemDisplayName>;
+  graphicalItem: {
+    props: Record<GraphicalItemShapeKey, unknown[]>;
+  };
+  itemData: unknown[];
+};
+
+function isFunnel(itemDisplayName: ItemDisplayName, _item: unknown): _item is FunnelItem {
+  return itemDisplayName === 'Funnel';
+}
+
+function isPie(itemDisplayName: ItemDisplayName, _item: unknown): _item is PieItem {
+  return itemDisplayName === 'Pie';
+}
+
+function isScatter(itemDisplayName: ItemDisplayName, _item: unknown): _item is ScatterItem {
+  return itemDisplayName === 'Scatter';
+}
```

Comment:
I can pass in `graphicalItem` instead of `itemDisplayName` and check the existence of the particular shape key(trapezoids, sector, points) inside of `graphicalItem.props`.

```typescript
function isFunnel(graphicalItem: GraphicalItem, _item: unknown): _item is FunnelItem {
  return 'trapezoids' in graphicalItem.props
}
```